crates/miden-objects/src/ecdsa_signer.rs

```diff
@@ -0,0 +1,29 @@
+use miden_core::Word;
+use miden_crypto::dsa::ecdsa_k256_keccak::SecretKey;
+
+use crate::crypto::dsa::ecdsa_k256_keccak as ecdsa;
+
+// ECDSA SIGNER
+// ================================================================================================
+
+/// Trait which abstracts the signing of ECDSA signatures. Used for signing block headers.
+///
+/// Production-level implementations will involve some sort of secure remote backend. The trait also
+/// allows for testing with local and ephemeral signers.
+pub trait EcdsaSigner {
+    fn sign(&self, message: Word) -> ecdsa::Signature;
+    fn public_key(&self) -> ecdsa::PublicKey;
+}
```

Comment:
I'm not sure this trait is needed, at least at this point (we may end up handling signing in a different way). So, I'd probably remove it.

If we do want to keep it, I'd move it into the `block` namespace, and probably rename into `BlockSigner` or something like that (i.e., the important part is not what signature we are using but that this is for signing blocks).